packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -255,6 +265,62 @@ export function buildThunks<
       return ret
     }
 
+  const upsertQueryData: UpsertQueryDataThunk<EndpointDefinitions, State> =
```

Comment:
I'm not sure if we need to go with the full "updateRecipe" here. I think it would also be fine to have the user say "this is the data, put it into the endpoint or overwrite everything that is there".
Or do you think it would be used a lot?

Reply:
It just felt the most natural to me. I don't necessarily know if the other query already has data or not, and this allows me to handle both cases easily. In cases where the data is only a primitive / simple value you can just return the new value from the recipe, but it is also possible to reuse data if the draft is already there. Otherwise I would need to manually check if there is already cached data or always specify both updateRecipe and upsertValue.